src/Runner.Worker/ExecutionContext.cs

```diff
@@ -982,6 +983,19 @@ public void QueueSummaryFile(string name, string filePath, Guid stepRecordId)
             _jobServerQueue.QueueResultsUpload(stepRecordId, name, filePath, ChecksAttachmentType.StepSummary, deleteSource: false, finalize: true, firstBlock: true, totalLines: 0);
         }
 
+        public void QueueDiagnosticLogFile(string name, string filePath)
+        {
+            ArgUtil.NotNullOrEmpty(name, nameof(name));
+            ArgUtil.NotNullOrEmpty(filePath, nameof(filePath));
+
+
+            if (!File.Exists(filePath))
+            {
+                throw new FileNotFoundException($"Can't upload diagnostic log file: {filePath}. File does not exist.");
+            }
+            _jobServerQueue.QueueResultsUpload(_record.Id, name, filePath, CoreAttachmentType.DiagnosticLog, deleteSource: false, finalize: true, firstBlock: true, totalLines: 0);
```

Comment:
can you create a new type? 🙇 
ex: `CoreAttachmentType.ResultDiagnosticLog = "Results.Core.DiagnosticLog"`